src/bos/operators/utils/clients/bos/base.py

```diff
@@ -121,3 +122,76 @@ def delete_items(self, **kwargs):
             return json.loads(response.text)
         else:
             return None
+
+
+class BaseBosTenantAwareEndpoint(BaseBosEndpoint):
+    """
+    This base class provides generic access to the BOS API for tenant aware endpoints.
+    The individual endpoint needs to be overridden for a specific endpoint.
+    """
+
+    @log_call_errors
+    def get_item(self, item_id, tenant):
+        """Get information for a single BOS item"""
+        url = self.base_url + '/' + item_id
+        session = requests_retry_session()
+        response = session.get(url, headers=get_tenant_header(tenant))
+        response.raise_for_status()
+        item = json.loads(response.text)
+        return item
+
+    @log_call_errors
+    def get_items(self, **kwargs):
+        """Get information for all BOS items"""
+        session = requests_retry_session()
+        headers = None
+        if "tenant" in kwargs:
+            headers = get_tenant_header(kwargs["tenant"])
+            del kwargs["tenant"]
+        response = session.get(self.base_url, params=kwargs, headers=headers)
+        response.raise_for_status()
+        items = json.loads(response.text)
+        return items
+
+    @log_call_errors
+    def update_item(self, item_id, tenant, data):
+        """Update information for a single BOS item"""
+        url = self.base_url + '/' + item_id
+        session = requests_retry_session()
+        response = session.patch(url, json=data, headers=get_tenant_header(tenant))
+        response.raise_for_status()
+        item = json.loads(response.text)
+        return item
+
+    @log_call_errors
+    def update_items(self, tenant, data):
+        """Update information for multiple BOS items"""
+        session = requests_retry_session()
+        response = session.patch(self.base_url, json=data, headers=get_tenant_header(tenant))
+        response.raise_for_status()
+        items = json.loads(response.text)
+        return items
+
+    @log_call_errors
+    def put_items(self, tenant, data):
+        """Put information for multiple BOS Items"""
+        session = requests_retry_session()
+        response = session.put(self.base_url, json=data, headers=get_tenant_header(tenant))
+        response.raise_for_status()
+        items = json.loads(response.text)
+        return items
+
+    @log_call_errors
+    def delete_items(self, **kwargs):
```

Comment:
Correct, because the base class also doesn't have a delete_item method, because we don't use it in the operators.  There's an argument to be made for completeness, but I would rather only have what we need, especially since aiming for completeness when this file was first written led to invalid methods that didn't match actual endpoints.

Reply:
Sounds good.